examples/decide_feature_flags.rs

```diff
@@ -0,0 +1,22 @@
+use posthog_rs::{Client, ClientOptionsBuilder, PublicAPI};
+
+async fn run() {
+    let client_options = ClientOptionsBuilder::new()
+        .set_endpoint(dotenv::var("POSTHOG_URL").unwrap())
+        .set_api_key(dotenv::var("POSTHOG_PROJECT_API_KEY").unwrap())
```

Comment:
We will have a clash of API keys here, as we cant use the same API key to run `decide` and `list_feature_flags` (as the former wants PROJECT_API_KEY, and the latter wants API_KEY)
 ... Easiest would probably be to allow both?

Would be nice to use some sort of [type state pattern](https://www.youtube.com/watch?v=_ccDqRTx-JU) to only allow the different API calls based on how the client is configured!

Reply:
Good point! So if the two types of API keys is given then all calls can be made. Or where you thinking that only one API key per client and you get the set of calls that the API key has permission to?

Reply:
Yeah something like:
```rust
struct PublicClient {...}
impl PublicAPI for PostHogPublicClient {...}

struct PrivateClient {...}
impl PrivateAPIParts for PostHogPrivateClient {...}

struct Client {...}
impl PublicAPI for Client {...}
impl [privateAPIParts] for Client {...}
```

And then the `ClientOptionsBuilder` could instead be a `ClientBuilder`

Reply:
I tried to do a variation of this. You can create the client as usual and it will default to Public and if you want a client to be private you set it as `    let client = Client::new(client_options).private();` and than you can call `let feature_flags = client
        .list_feature_flags(..)...` for example. 
It relies on the creator of Client to set it state of it, so it might not be the most elegant solution.